skrub/tests/test_deduplicate.py

```diff
@@ -95,3 +97,57 @@ def test__create_spelling_correction(seed: int = 123) -> None:
             spelling_correction.values[clusters == n].astype("int")
             == counts[clusters == n].max()
         ).all()
+
+
+@skip_if_no_parallel
+def test_parallelism():
+    # Test that parallelism works
+    X = make_deduplication_data(examples=['black', 'white'],
+                                entries_per_example=[15, 15])
+    y = deduplicate(X, n_jobs=None)
+    for n_jobs in [2, 1, -1]:
+        y_parallel = deduplicate(X, n_jobs=n_jobs)
+        assert_array_equal(y, y_parallel)
+
+    # Test with threading backend
+    with joblib.parallel_backend("threading"):
+        y_threading = deduplicate(X, n_jobs=1)
+    assert_array_equal(y, y_threading)
```

Comment:
I feel like this test doesn't really test the parallelization: the function could ignore the `n_jobs` parameter, and the test would pass anyway.
I think we should actually time the calls with different `n_jobs` and assert that a higher number of jobs would make the call faster.

Reply:
Ok, I see what you mean, but a higher number of jobs doesn't always make the call faster.
It does so when the data is big enough, which is why we need to add it. But adding such an example to the test suite may make our tests super slow to run each time, so I don't know if this is a good solution..